### ISSUE TITLE
[Question] Multi-objective auto-sklearn?

### ISSUE THREAD

**Issue by wd2311** [automl/auto-sklearn#1317]
Are there any multi-objective optimization capabilities in auto-sklearn?

I use AutoML, and most of the time, I am not interested in differences of 1-2% in a performance metric, but care about things like:  Minimizing the number of input features, building simple models that are highly interpretable, or building models with a fast inference speed.

My question is - are there any ongoing efforts to run a model search optimizing not just for performances, but for simple (e.g. low feature, low parameter) models?  or fair models?  or fast inference speed models?  If not, I wonder if anybody building auto-sklearn would be interested in discussing the concept.

-Will David, wdavid2222@gmail.com

**Comment by eddiebergman** [comment]
Hi @wd2311,

Currently we rely on `SMAC` which is a single objective optimizer. Many of the things you mention are not directly a metric calculable from the objective function that can be passed. While we are looking at more flexible pipelines (to include things such as fairness based steps) and also looking at a more generic optimizer interface (such as allowing multi-objective functions, this is not something we can currently support.

I will leave this issue open in the mean time and any further comments or discussion would be highly appreciated.

Best,
Eddie

**Comment by rabsr** [comment]
Hi @eddiebergman 

Recently SMAC has included  [mutli-objective function](https://github.com/automl/SMAC3/pull/801) capability and is expected to release soon in [version 1.2](https://github.com/automl/SMAC3/pull/816). I am wondering if this enhancement is still relevant to scope of auto-sklearn. I can explore more on this and contribute to the project.

**Comment by mfeurer** [comment]
Working on this in #1455.

**Comment by mfeurer** [comment]
This will become available in the upcoming 0.15 release and we just released a first dummy ensemble via #1523.
